server/src/main/java/org/opensearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -556,10 +556,15 @@ private void processLeafFromQuery(LeafReaderContext ctx, Sort indexSortPrefix) t
     }
 
     @Override
-    protected LeafBucketCollector getLeafCollector(LeafReaderContext ctx, LeafBucketCollector sub) throws IOException {
-        boolean optimized = filterRewriteOptimizationContext.tryOptimize(ctx, this::incrementBucketDocCount, segmentMatchAll(context, ctx));
-        if (optimized) throw new CollectionTerminatedException();
+    protected boolean tryPrecomputeAggregationForLeaf(LeafReaderContext ctx) throws IOException {
+        if (subAggregators().length == 0) {
```

Comment:
I am wondering if this was not handled before? If not, should be a bug?

Reply:
Hmm... I don't recall why I added this check. Let me try removing it.

It might not be necessary...

Reply:
Hmm... I don't recall why I added this check. Let me try removing it. It might not be necessary.

Reply:
As is, I think the fact that we were skipping the call to `finishLeaf()` when precomputing a segment was a bug.

Just because we're precomputing *this* segment doesn't mean we don't need to call `finishLeaf` on the previous segment.

Reply:
I removed the `subAggregators` check and the tests still pass, so I probably should not have added it in the first place.